File: Python/StringManipulation.py
def count_characters(input_str):
    return len(input_str)

def find_substrings(input_str, substring):
    return input_str.count(substring)

def reverse_words(input_str):
    words = input_str.split()
    reversed_words = [word[::-1] for word in words]
    return ' '.join(reversed_words)


print(count_characters("Hello, World!"))  # Output: 13
print(find_substrings("Hello, World! Hello, Python!", "Hello"))  # Output: 2
print(reverse_words("Hello, World!"))  # Output: "olleH, dlroW!"

File: Rust/StringManipulation.rs
fn count_characters(input: &str) -> usize {
    input.len()
}

fn find_substrings(input: &str, substring: &str) -> usize {
    input.matches(substring).count()
}

fn reverse_words(input: &str) -> String {
    let words: Vec<&str> = input.split_whitespace().collect();
    let reversed_words: Vec<String> = words.iter().map(|word| word.chars().rev().collect()).collect();
    reversed_words.join(" ")
}

fn main() {
    let input_str = "Hello, World!";
    println!("{}", count_characters(input_str)); // Output: 13

    // Reverse word call
    input_str = "Hello, World!";
    println!("{}", reverse_words(input_str)); // Output: "olleH, dlroW!"

    // Find Substring call
    input_str = "Hello, World! Hello, Rust!";
    let substring = "Hello";
    println!("{}", find_substrings(input_str, substring)); // Output: 2
}